src/writer.rs

```diff
@@ -63,11 +63,25 @@ impl<W: Write> BzCompressor<W> {
         }
         Ok(self.w.take().unwrap())
     }
+
+    /// Returns the number of bytes produced by the compressor
+    pub fn total_out(&self) -> u64 {
+        self.stream.total_out()
+    }
+
+    /// Returns the number of bytes consumed by the compressor
+    pub fn total_in(&self) -> u64 {
+        self.stream.total_in()
+    }
 }
 
 impl<W: Write> Write for BzCompressor<W> {
     fn write(&mut self, data: &[u8]) -> io::Result<usize> {
-        self.do_write(data, Action::Run)
+        if data.len() != 0 {
```

Comment:
How come this branch was necessary? The `do_write` function already has to handle 0-length data vectors for flush below, so I'm curious what the failure mode here was